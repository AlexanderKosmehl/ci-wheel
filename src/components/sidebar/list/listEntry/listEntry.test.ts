import {
  describe, expect, it, vi,
} from 'vitest';
import generateListEntry from './listEntry';

describe('generateListEntry', () => {
  const listEntry = generateListEntry({
    label: 'Test',
    isDone: false,
  });

  it('generates component correctly', () => {
    expect(listEntry).toMatchSnapshot();
  });

  it('adds isDone changes', () => {
    const doneEntry = generateListEntry({
      label: 'Test',
      isDone: true,
    });

    expect(doneEntry).toMatchSnapshot();
  });
});
