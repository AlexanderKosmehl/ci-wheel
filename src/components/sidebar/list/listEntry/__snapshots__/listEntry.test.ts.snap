// Vitest Snapshot v1

exports[`generateListEntry > adds isDone changes 1`] = `
<li
  class="_container_1761de"
  style="order: 1;"
>
  <span
    class="_label_1761de _isDone_1761de"
    data-test="listEntryLabel"
  >
    Test
  </span>
  <div
    class="_buttonWrapper_1761de"
  >
    <button
      class="_button_7c1f96 _button_1761de"
      data-test="listEntryReAddButton"
    >
      <img
        class="_icon_7c1f96"
        src="/src/assets/add-icon.svg"
      />
    </button>
    <button
      class="_button_7c1f96 _button_1761de"
      data-test="listEntryDeleteButton"
    >
      <img
        class="_icon_7c1f96"
        src="/src/assets/delete-icon.svg"
      />
    </button>
  </div>
</li>
`;

exports[`generateListEntry > generates component correctly 1`] = `
<li
  class="_container_1761de"
>
  <span
    class="_label_1761de"
    data-test="listEntryLabel"
  >
    Test
  </span>
  <div
    class="_buttonWrapper_1761de"
  >
    <button
      class="_button_7c1f96 _button_1761de"
      data-test="listEntryDeleteButton"
    >
      <img
        class="_icon_7c1f96"
        src="/src/assets/delete-icon.svg"
      />
    </button>
  </div>
</li>
`;
